src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs

```diff
@@ -1094,53 +1096,71 @@ internal static StackValue[] GenerateRangeByAmount(decimal start, long amount, d
                 return StackValue.Null;
             }
 
-            double startValue = svStart.ToDouble().RawDoubleValue;
-            if (double.IsInfinity(startValue) || double.IsNaN(startValue))
+            StackValue[] range = null;
+
+            if (svStart.IsNumeric)
```

Comment:
In row 1052 we checked, that `svStart` and `svEnd` are Int or String values. So here we can be sure if `svStart` is numeric, then `svEnd` is numeric as well.

So, if incoming value is numeric, we call `GenerateNumericRange`. If incoming value is string, we call `GenerateAlphabetRange`.